cli/host.go

```diff
@@ -103,10 +107,42 @@ func (cmd *HostStatusCommand) Execute(_ []string) error {
 // HostTerminateCommand is the subcommand to terminate a host
 type HostTerminateCommand struct {
 	GlobalOpts *Options `no-flag:"true"`
-	HostID     string   `short:"h" long:"host" description:"terminates the specified host" required:"true"`
+	HostID     string   `long:"host" short:"h" description:"terminates the specified host" required:"true"`
 }
 
-// Execute will...
+// Execute terminates a given host
 func (cmd *HostTerminateCommand) Execute(_ []string) error {
-	return errors.New("not implemented")
+	if cmd.HostID == "" {
+		return errors.New("host ID cannot be blank")
+	}
+
+	client, _, _, err := getAPIClients(cmd.GlobalOpts)
+	if err != nil {
+		return err
+	}
+
+	data := struct {
+		HostID string `json:"host_id"`
+		Action string `json:"action"`
+	}{cmd.HostID, "terminate"}
+
+	rPipe, wPipe := io.Pipe()
+	encoder := json.NewEncoder(wPipe)
+	go func() {
```

Comment:
Why is this in a separate goroutine?

Reply:
This was copied from other CLI code that formed request bodies. I tried it in the main goroutine but that caused the encode method to block indefinitely